pkg/paillier/public.go

```diff
@@ -64,30 +76,35 @@ func (pk PublicKey) Enc(m *big.Int) (*Ciphertext, *big.Int) {
 //
 // ct = (1+N)ᵐρᴺ (mod N²)
 func (pk PublicKey) EncWithNonce(m, nonce *big.Int) *Ciphertext {
-	if m.CmpAbs(pk.nHalf) == 1 {
-		panic("paillier.Encrypt: tried to encrypt message outside of range [-(N-1)/2, …, (N-1)/2]")
```

Comment:
Returning an error would be wise, as otherwise the decryption would be wrong, but for now it this is fine!

Reply:
Actually, if you follow the call chain, this condition gets checked later when we do `m.CheckInRange` inside of `encWithNonceNat`